modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/nativelib/ArtifactCopyNativeLibraryFinder.java

```diff
@@ -113,26 +111,37 @@ private File copyNativeLibrary(String name, String libraryPath) {
     }
   }
 
-  private String findLibraryLocally(String name) {
-    String nativeLibName = System.mapLibraryName(name);
-    String extension = FilenameUtils.getExtension(nativeLibName);
-    Optional<URL> nativeLib =
-        stream(urls).filter((URL url) -> {
-          String fullPath;
-          try {
-            fullPath = url.toURI().toString();
-          } catch (URISyntaxException e) {
-            return false;
-          }
-          String baseName = getBaseName(fullPath);
-          return baseName.contains(name) && (fullPath.endsWith(extension) || IS_OS_MAC && fullPath.endsWith(JNILIB_EXTENSION));
-        }).findFirst();
+  private String findLibraryLocally(String libraryName) {
+    String localLibraryNameWeakMatch = null;
 
-    if (nativeLib.isPresent()) {
-      return nativeLib.get().getFile();
-    } else {
-      return null;
+    for (URL url : this.urls) {
+      try {
+        String localLibraryNameFullPath = url.toURI().toString();
+        String extension = getExtension(System.mapLibraryName(libraryName));
+        if (endsWithExtension(localLibraryNameFullPath, extension)) {
+          if (isAStrictMatch(libraryName, localLibraryNameFullPath)) {
+            return url.getFile();
+          } else if (localLibraryNameWeakMatch == null && isAWeakMatch(libraryName, localLibraryNameFullPath)) {
```

Comment:
Please add a comment prior to this line indicating that this is to maintain backwards compatibility with the previous behavior